### DISCUSSION TITLE
blog/fluentvalidation-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#8] [General]
# blog/fluentvalidation-in-aspnet-core/

Learn how to make powerful validations in your ASP.NET Core applications using FluentValidation. When it comes to Validating Models and incoming HTTP Requests, aren't we all leaning toward Data Annotations? Although it can be quick to set up Data Annotations on your models, there are a few drawbacks to this approach.

https://codewithmukesh.com/blog/fluentvalidation-in-aspnet-core/?lidx=13&wpid=471671

**Comment by nileshgr** [comment]
Thanks for the nice intro to FluentValidation!

**Comment by abelmiraval** [comment]
Hello, how to validate a decimal type field that does not send in the request?

**Comment by zabronm** [comment]
This is another excellent read from you, thank you.

**Comment by afaolek** [comment]
This is a lovely read on `FluentValidation`. However, I'm in this situation, which is what brought me to this article.
In my custom validator, I need to access some values from  repository. First, I get an object from the repository using `IRepository.GetByIdAsync(id)`. As you can guess, it is an `async` method. From this object, I retrieve two properties as values, max and min, which I need for the validation. I can use dependency injection to get the repository in the constructor, which is also where the validation rules are defined. The problem now is, I can't call the async method in the constructor to retrieve those values.
What do I do?

**Comment by codedcozy12** [comment]
This is a very nice thread about fluent validation i understand everything now thanks very much

**Comment by neo2k28** [comment]
Your site is awesome!
